rfc/build-system/build-system.md

```diff
@@ -0,0 +1,698 @@
+
+Commit timestamps aren't guaranteed to increase monotonically, so technically
+Maven won't always sort these versions correctly. If, for example, you're
+developing a plugin and add a dependency on the latest version of echo, you may
+not resolve the _actual_ latest version. Because we generally use squash
+commits, however, this shouldn't affect us much.
+
+The advantage of using commit timestamp (instead of build timestamp) is that you
+can determine the name of an artifact using only the branch and commit log; if
+we used build time, you'd have to list all the artifacts in the repository and
+try to match the short SHA.
+
+On master, the first part of the artifact version ("2020.4") will be determined
+by looking at the existing branches and calculating the expected next release.
+If the latest release branch is `release-2021.2.x`, we'll use 2021.3 for the
+main artifact version.
+
+### Releases
+
+GitHub Actions workflows are primarily triggered by events on a GitHub
+repository. To facilitate triggering these workflows, we will create a
+`spinnaker/release-operations` repository. Various actions will be performed on
+this repository to launch the release-related workflows. Write access to this
+repository will be limited to the `release-managers` GitHub team, so only those
+members will have access to launch the workflows.
+
+#### Create Release Branches
+
+To create the release branches for a Spinnaker release, a release manager will
+create a branch in the `release-operations` repository. The commit used for the
+branch is ignored. We will continue to use the same release branches we do now:
+a 2020.4 release is backed by a branch named `release-2020.4.x`.
+
+Creating a `release-*` branch triggers a workflow that will create a branch with
+the same name in all release-relevant repositories.
+
+#### Create a Testing Release
+
+To create a Spinnaker release for testing, tag a commit in the
+`release-operations` repository. The commit being tagged is ignored. The tag
+should start with a branch name like `master-*` or `release-2020.4.x-*`.
+Generally speaking, these will be date-based: `master-20200614-220719`. (Using a
+tag to trigger a release is admittedly weird, but it will generally only be done
+via a Jenkins job and will go away entirely when we eventually remove Jenkins.)
+
+It will trigger a build on all 11 microservices to publish Debian and Docker
+images to the artifact repository. Once all those builds have succeeded, the BOM
+will be published and containers tagged with this release version.
+
+#### Publishing a Release
+
+To publish a release of Spinnaker, a release manager will create a GitHub
+release in the `release-operations` repository. The tag used for the release is
+the testing release that will be published.
+
+All published artifacts from the testing release will be updated with the new
+version:
+
+- BOM (for Halyard deployments): The existing BOM will be copied, with the
```

Comment:
If all services use the same version, we don't need a BOM anymore. `spinnaker-monitoring` and `deck` being the exceptions. Is the plan to eventually align these versions - perhaps just tagging the produced images? Managing BOMs is not fun in some environments.

Reply:
As of this latest release, we actually do tag the containers! `gcr.io/spinnaker-marketplace/deck:spinnaker-1.21.0`, etc. Because of the aforementioned version conflicts, we do have to prepend `spinnaker-` to the tags, since `deck:1.21.0` is something else entirely...

Anyway, that's what I meant (perhaps unclearly) two lines below:

> - Docker: the existing containers for each microservice will be tagged with the new version numbers

It's true that we don't really need the BOMs for containers now (but still do for Debian packages), but I don't really want to bother fixing Halyard to not use them, so they'll probably remain a thing until both Halyard and the Debian packages are fully deprecated.

Reply:
I updated the bullet to have more context around this, thanks:

> - BOM (for Halyard deployments): The existing BOM will be copied, with the
  included `version` field updated to reflect the new version number.
  (Deployments that aren't using Halyard will just fetch the containers
  directly, but the BOM will continue to be written until Halyard is no longer
  a supported tool.)